source/mir/sparse/package.d

```diff
@@ -0,0 +1,614 @@
+/++
+$(H2 Sparse Tensors)
+
+This is a submodule of $(LINK2 mir_ndslice.html, mir.ndslice).
+
+
+License:   $(WEB www.boost.org/LICENSE_1_0.txt, Boost License 1.0).
+
+Authors:   Ilya Yaroshenko
+
+Source:    $(PHOBOSSRC std/_experimental/_ndslice/_sparse.d)
+
+Macros:
+SUBMODULE = $(LINK2 mir_ndslice_$1.html, mir.ndslice.$1)
+SUBREF = $(LINK2 mir_ndslice_$1.html#.$2, $(TT $2))$(NBSP)
+T2=$(TR $(TDNW $(LREF $1)) $(TD $+))
+T4=$(TR $(TDNW $(LREF $1)) $(TD $2) $(TD $3) $(TD $4))
+
++/
+module mir.sparse;
+
+import std.traits;
+import std.meta;
+
+import mir.ndslice.slice;
+
+private enum isIndex(I) = is(I : size_t);
+
+//TODO: replace with `static foreach`
+private template Iota(size_t i, size_t j)
+{
+    static assert(i <= j, "Iota: i should be less than or equal to j");
+    static if (i == j)
+        alias Iota = AliasSeq!();
+    else
+        alias Iota = AliasSeq!(i, Iota!(i + 1, j));
+}
+
+/++
+Sparse tensors represented in Dictionary of Keys (DOK) format.
+
+Params:
+    N = dimension count
+    lengths = list of dimension lengths
+Returns:
+    `N`-dimensional slice composed of indexes
+See_also: $(LREF Sparse)
++/
+Sparse!(Lengths.length, T) sparse(T, Lengths...)(Lengths lengths)
+    if (allSatisfy!(isIndex, Lengths))
+{
+    return .sparse!(T, Lengths.length)([lengths]);
+}
+
+/// ditto
+Sparse!(N, T) sparse(T, size_t N)(auto ref size_t[N] lengths)
+{
+    import mir.ndslice.slice: sliced;
+    T[size_t] table;
+    table[0] = 0;
+    table.remove(0);
+    assert(table !is null);
```

Comment:
that's is still so weird - it really does make a difference (I just checked myself). Could you please leave a comment there - probably the people at phobos will complain otherwise too ;-)